contracts/utils/cryptography/draft-ERC7739Signer.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IERC1271} from "../../interfaces/IERC1271.sol";
+import {EIP712} from "./EIP712.sol";
+import {MessageHashUtils} from "./MessageHashUtils.sol";
+import {ERC7739Utils} from "./draft-ERC7739Utils.sol";
+import {ShortStrings} from "../ShortStrings.sol";
+
+/**
+ * @dev Validates signatures wrapping the message hash in a nested EIP712 type. See {ERC7739Utils}.
+ *
+ * Linking the signature to the EIP-712 domain separator is a security measure to prevent signature replay across different
+ * EIP-712 domains (e.g. a single offchain owner of multiple contracts).
+ *
+ * This contract requires implementing the {_validateSignature} function, which passes the wrapped message hash,
+ * which may be either an typed data or a personal sign nested type.
+ *
+ * NOTE: {EIP712} uses {ShortStrings} to optimize gas costs for short strings (up to 31 characters).
+ * Consider that strings longer than that will use storage, which may limit the ability of the signer to
+ * be used within the ERC-4337 validation phase (due to ERC-7562 storage access rules).
```

Comment:
Is this accurate? The account is able to read its own storage during validation so I don't think there will be any issues.

```suggestion
```

Reply:
```suggestion
 *
 * NOTE: {EIP712} uses {ShortStrings} to optimize gas costs for short strings (up to 31 characters). Consider that 
 * strings longer than that, and strings of any size for the upgradable variants of this contract, will be placed in 
 * storage. This may limit the ability of the signer to be used within the ERC-4337 validation. If the sender of the
 * user operation performs an ERC-1271 check against this signer, the ERC-7562 storage access rules will be broken.
```

Reply:
> That is in case account A relies on another other contract B for validation:

Yes but this is a much larger issue than EIP-712 name and version strings, because the public keys will usually be in storage.

IMO mentioning this here is a distraction.

Reply:
There are ways to put the public keys outside of storage (for example using a Proxy with immutable args) ... but the dependency on ERC712 prevents is just too much.

I don't have a strong opinion either way :/

I'm considering merging with (or without) the message for the audit, and have PR with all the proposed changes (in this case adding or removing) that we'll discuss in parallele of the audit.